File: badmin/badmin.cocktail/src/main/java/nl/tumbolia/bar/badmin/cocktail/Application.java

package nl.tumbolia.bar.badmin.cocktail;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

/**
 * @author Erwin.van.den.Beld
 *
 */
@SpringBootApplication
public class Application implements CommandLineRunner {

	@Autowired
	private CocktailRepository repository;

	public static void main(String[] args) {
		SpringApplication.run(Application.class, args);
	}

	@Override
	public void run(String... args) throws Exception {

		repository.deleteAll();

		// save a couple of customers
		repository.save(new Cocktail("Gin-fizz", "Smith"));
		repository.save(new Cocktail("B-52", "Smith"));

		// fetch all customers
		System.out.println("Cocktail found with findAll():");
		System.out.println("-------------------------------");
		for (Cocktail cocktail : repository.findAll()) {
			System.out.println(cocktail);
		}
		System.out.println();

		// // fetch an individual customer
		// System.out.println("Customer found with findByFirstName('Alice'):");
		// System.out.println("--------------------------------");
		// System.out.println(repository.findByFirstName("Alice"));
		//
		// System.out.println("Customers found with findByLastName('Smith'):");
		// System.out.println("--------------------------------");
		// for (Customer customer : repository.findByLastName("Smith")) {
		// System.out.println(customer);
		// }

	}

}

File: badmin/badmin.cocktail/src/main/java/nl/tumbolia/bar/badmin/cocktail/application.kt
package nl.tumbolia.bar.badmin.cocktail

import org.springframework.boot.CommandLineRunner
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.annotation.Bean
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestController
import javax.validation.Valid

@SpringBootApplication
open class Application  {
	
	companion object {
		@JvmStatic fun main(args: Array<String>) {
			SpringApplication.run(Application::class.java, *args)
		}
	}		
}